opm/simulators/flow/TemperatureModel.hpp

```diff
@@ -248,24 +256,55 @@ class TemperatureModel : public GenericTemperatureModel<GetPropType<TypeTag, Pro
+        Scalar sum_pv = 0.0;
+        Scalar sum_pv_not_converged = 0.0;
         for (const auto& elem : elements(simulator_.gridView(), Dune::Partitions::interior)) {
             unsigned globI = elemMapper.index(elem);
-            maxNorm = max(maxNorm, std::abs(this->energyVector_[globI]));
+            const auto pvValue =  simulator_.problem().referencePorosity(globI, /*timeIdx=*/0)
+            *  simulator_.model().dofTotalVolume(globI);
+
+            if (!isNumericalAquiferCell(elem)) {
+                if (std::abs(this->energyVector_[globI]) > tolerance_cnv_energy) {
+                    sum_pv_not_converged += pvValue;
+                }
+                sum_pv += pvValue;
+            }
+            maxNorm = max(maxNorm, std::abs(this->energyVector_[globI])/ pvValue);
             sumNorm += std::abs(this->energyVector_[globI]);
         }
         maxNorm = simulator_.gridView().comm().max(maxNorm);
         sumNorm = simulator_.gridView().comm().sum(sumNorm);
-        const int globalNumCells = simulator_.gridView().comm().sum(numCells);
-        sumNorm /= globalNumCells;
-        const auto tolerance_cnv_energy = Parameters::Get<Parameters::ToleranceCnvEnergy<Scalar>>();
-        const auto tolerance_energy_balance = Parameters::Get<Parameters::ToleranceEnergyBalance<Scalar>>();
-        if (maxNorm < tolerance_cnv_energy || sumNorm < tolerance_energy_balance) {
-            const auto msg = fmt::format("Temperature model (TEMP): Newton converged after {} iterations", iter);
-            OpmLog::debug(msg);
+        sum_pv = simulator_.gridView().comm().sum(sum_pv);
+        sum_pv_not_converged = simulator_.gridView().comm().sum(sum_pv_not_converged);
+
+        // make the norm time and pore-volume invariant
+        sumNorm *= dt;
+        sumNorm /= sum_pv;
+
+        // make the norm time invariant (it is already divided by the pore-volume)
+        maxNorm *= dt;
+        Scalar relaxed_max_pv_fraction = Parameters::Get<Parameters::RelaxedMaxPvFraction<Scalar>>();
+        // Use relaxed tolerance if the fraction of unconverged cells porevolume is less than relaxed_max_pv_fraction
+        bool relax = (sum_pv_not_converged / sum_pv) <  relaxed_max_pv_fraction;
+        const auto tolerance_energy_balance = relax? Parameters::Get<Parameters::ToleranceEnergyBalanceRelaxed<Scalar>>():
+                                            Parameters::Get<Parameters::ToleranceEnergyBalance<Scalar>>();
+        tolerance_cnv_energy = relax? Parameters::Get<Parameters::ToleranceCnvEnergyRelaxed<Scalar>>():
+                                            Parameters::Get<Parameters::ToleranceCnvEnergy<Scalar>>();
+
+        const auto msg = fmt::format("Temperature model (TEMP): Newton iter {}: "
+                                     "CNV(E): {:.1e}, EB: {:.1e}",
+                                     iter, maxNorm, sumNorm);
+        OpmLog::info(msg);
+        if (maxNorm < tolerance_cnv_energy && sumNorm < tolerance_energy_balance) {
```

Comment:
I forgot to change tot ::debug here and maybe also below. I don't think the user would like all this information.